plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -1014,19 +1014,23 @@ protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCon
     protected void verifySchemaName(DatabaseMetaData databaseMetadata, String schemaName)
             throws SQLException
     {
+        // CockroachDB may return negative values
         // PostgreSQL truncates schema name to 63 chars silently
-        if (schemaName.length() > databaseMetadata.getMaxSchemaNameLength()) {
-            throw new TrinoException(NOT_SUPPORTED, format("Schema name must be shorter than or equal to '%s' characters but got '%s'", databaseMetadata.getMaxSchemaNameLength(), schemaName.length()));
+        int maxSchemaNameLength = databaseMetadata.getMaxSchemaNameLength();
+        if (maxSchemaNameLength > 0 && schemaName.length() > maxSchemaNameLength) {
+            throw new TrinoException(NOT_SUPPORTED, format("Schema name must be shorter than or equal to '%s' characters but got '%s'", maxSchemaNameLength, schemaName.length()));
         }
     }
 
     @Override
     protected void verifyTableName(DatabaseMetaData databaseMetadata, String tableName)
             throws SQLException
     {
+        // CockroachDB may return negative values
         // PostgreSQL truncates table name to 63 chars silently
-        if (tableName.length() > databaseMetadata.getMaxTableNameLength()) {
-            throw new TrinoException(NOT_SUPPORTED, format("Table name must be shorter than or equal to '%s' characters but got '%s'", databaseMetadata.getMaxTableNameLength(), tableName.length()));
+        int maxTableNameLength = databaseMetadata.getMaxTableNameLength();
+        if (maxTableNameLength > 0 && tableName.length() > maxTableNameLength) {
```

Comment:
We know that PostgreSQL will happily truncate the name instead of rejecting it, so these checks are especially important for PostgreSQL. It's not safe to skip the check when we lack the information.
What about doing an explicit check for CockroachDB, perhaps via `java.sql.DatabaseMetaData#getDatabaseProductName`.

Still, i think a dedicated CockroachDB is the way to go.

Reply:
[The docs](https://docs.oracle.com/javase/9/docs/api/java/sql/DatabaseMetaData.html) for `databaseMetadata.getMaxTableNameLength()` say that:
```
Returns:
    the maximum number of statements that can be open at one time; a result of zero means that there is no limit or the limit is not known
```
so I think it makes sense to check for zero.

Reply:
But the point is that the PostgreSQL JDBC Driver doesn't know it's talking to CockroachDB so it'll more likely than not return incorrect values from DatabaseMetaData (as happenned here).

I agree we shouldn't "fix" something which worked only coincidentally. This is similar to when MySQL connector broke MariaDB compatibility.

Reply:
Unfortunately, `PgDatabaseMetaData.getDatabaseProductName` returns hard coded string `PostgreSQL`. I couldn't find the smart way to identify the server type.